torch/csrc/autograd/python_variable_indexing.cpp

```diff
@@ -247,6 +247,76 @@ static PyObject* applyBoolGetitem(const Variable& self, bool index) {
   }
 }
 
+static bool canDispatchToLegacyIndexing(const Variable& self, const variable_list& vars) {
+  // mask indexing
+  if (vars.size() == 1 && vars[0].type().scalarType() == ScalarType::Byte && vars[0].is_same_size(self)) {
+    return true;
+  }
+
+  // single tensor indexing
+  int num_defined_variables = 0;
+  for (auto& variable : vars) {
+    auto is_defined = variable.defined();
+    num_defined_variables += is_defined;
+    if (is_defined && (variable.dim() != 1 || variable.type().scalarType() != ScalarType::Long || variable.numel() == 0)) {
```

Comment:
what do undefined variables mean here?

Reply:
I was unsure what properties undefined tensors could have, so I was just being overly cautious here. Do undefined tensors have a numel == 0?

Reply:
And the undefined tensors come from the fact that slices or eclipses append undefined tensors into the indexing list

Reply:
undefined tensors basically throw exceptions if you try to do operations on them.

Reply:
So this check is necessary, as we can have undefined tensors in the list